spec/datadog/core/remote/client_spec.rb

```diff
@@ -588,6 +588,24 @@
               end
             end
 
+            context 'with remote extra_services setting' do
+              it 'returns client_tracer' do
+                expect(Datadog.configuration.remote).to receive(:service).and_return('foo').at_least(:once)
```

Comment:
I don't quite get why this is necessary.

Without it it fails with:

```
  1) Datadog::Core::Remote::Client#sync #payload no sync errors client client_tracer with remote extra_services setting returns client_tracer
     Failure/Error: expect(client_payload[:client_tracer].tap { |h| h.delete(:tags) }).to eq(expected_client_tracer)
     
       expected: {:env=>nil, :extra_services=>["foo"], :language=>"ruby", :runtime_id=>"803326d7-3898-48eb-8e73-c4b2854fa7df", :service=>nil, :tracer_version=>"1.14.0"}
            got: {:env=>nil, :extra_services=>["foo"], :language=>"ruby", :runtime_id=>"803326d7-3898-48eb-8e73-c4b2854fa7df", :service=>"rspec", :tracer_version=>"1.14.0"}
     
       (compared using ==)
     
       Diff:
       @@ -2,6 +2,6 @@
        :extra_services => ["foo"],
        :language => "ruby",
        :runtime_id => "803326d7-3898-48eb-8e73-c4b2854fa7df",
       -:service => nil,
       +:service => "rspec",
        :tracer_version => "1.14.0",
```

Reply:
`rspec` is the fallback service name: https://github.com/DataDog/dd-trace-rb/blob/ff32e137b07d261d10179ea22176a875654521a7/lib/datadog/core/environment/ext.rb#L26

and it might be coming from here: https://github.com/DataDog/dd-trace-rb/blob/ff32e137b07d261d10179ea22176a875654521a7/lib/datadog/tracing/tracer.rb#L51

Reply:
What I don't get is why it consistently works for the other spec examples but consistently not for this one 🤔

Reply:
What I don't get is why it consistently works for the other spec examples but consistently not for this one 🤔